### PR TITLE
Refresh: update Cloud free trial doc for new flow

### PR DESCRIPTION
**Pull request by matthijskrul**
## Summary of the changes / Why this is an improvement
- Updates the Cloud free trial documentation for the release of the free trial webpage form
NB: Link to regular subscription (Stripe route) tutorial to be added once that is released

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/free-trial.rst [hunk 1]
```diff
@@ -5,14 +5,17 @@ Free trial
...
+subscription tutorial, and to choose the Development subscription plan. As a
+secondary option, you can also refer to our SaaS marketplace offers on `Azure
+Marketplace`_ and `AWS Marketplace`_ and choose the Development plan there.
+(For more information on what that means, see our `reference on subscription
+plans`_).
+
+A different approach is to avail yourself of the possibility of our free trial:
+a fully functioning CrateDB Cloud cluster running for 14 days without cost, so
+you can experience the benefits of CrateDB Cloud without any immediate
+commitment.
```
Comment (justien) [inline docs/free-trial.rst]:
```suggestion
this. One is to subscribe to CrateDB Cloud by following our direct
subscription tutorial, and to choose the Development subscription plan.
Another option is via our SaaS marketplace offers on `Azure Marketplace`_ 
and `AWS Marketplace`_ and choose the Development plan there. For more
information on the differences, see our `reference on subscription plans`_.

A third option is our free trial: a fully functioning CrateDB Cloud cluster
running for 14 days without cost, so you can experience the benefits of
CrateDB Cloud without any immediate commitment.
```

File: docs/free-trial.rst [hunk 2]
```diff
@@ -27,11 +30,20 @@ the deployment process, as will be shown in the following steps.
...
 =======
 
-In order to access CrateDB Cloud, you must first sign up with an account.
-Simply follow the steps in the :ref:`tutorial for signing up <sign-up>`. When
-you have successfully signed up, you will be forwarded to the CrateDB Cloud
-Console subscription overview. It is from here that you can make use of the
-free trial option. The first step is to set up an organization.
+In order to access the free trial, you must first sign up with an account.
+The signup process starts at the webpage form. Simply fill out the form and
+click *Submit* to proceed.
```
Comment (justien) [inline docs/free-trial.rst]:
```suggestion
In order to access the free trial, you must first sign up with an account.
The signup process starts at the webpage form. Fill out the form and
click *Submit* to proceed.
```

File: docs/free-trial.rst [hunk 3]
```diff
@@ -27,11 +30,20 @@ the deployment process, as will be shown in the following steps.
...
+.. image:: _assets/img/free-trial-webform.png
+   :alt: Free trial webform
+
+You will see a forwarding screen confirming your information was received. This
+will forward you to the signup process for creating a username and password for
+CrateDB Cloud (supported by Cognito). Follow the steps and log in (or if you
+already have a username and password for CrateDB Cloud, simply log in now).
+When you have successfully logged in, you will be forwarded to the CrateDB
+Cloud Console subscription overview. It is from here that you can make use of
+the free trial option. The first step is to set up an organization.
```
Comment (justien) [inline docs/free-trial.rst]:
```suggestion
You will see a forwarding screen confirming your information was received. This
takes you to the signup process where you can create a username and password for
CrateDB Cloud (supported by Cognito). Follow the steps and log in (or if you
already have a username and password for CrateDB Cloud, log in with those credentials).
Once logged in, you will be forwarded to the CrateDB Cloud Console subscription overview.
where you will find the free trial option. The first step is to set up an organization.
```

**Review by justien — APPROVED** [review]
A few suggestions that reduce the formality of the language a little bit.

**Review by carlotasoto — APPROVED** [review]

**Review by infoverload — APPROVED** [review]

**Pull request merged**